plugin/http/httptrace/httptrace.go

```diff
@@ -59,10 +62,38 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := t.base().RoundTrip(req)
 
 	// TODO(jbd): Add status and attributes.
-	trace.EndSpan(ctx)
+	if err != nil {
+		trace.EndSpan(ctx)
+		return resp, err
+	}
+
+	// trace.EndSpan(ctx) will be invoked after
+	// resp.Body.Close() has been invoked.
+	resp.Body = &bodyProxyReader{rc: resp.Body, ctxToEnd: ctx}
 	return resp, err
 }
 
+type bodyProxyReader struct {
+	rc        io.ReadCloser
+	closeOnce sync.Once
+	ctxToEnd  context.Context
+}
+
+var _ io.ReadCloser = (*bodyProxyReader)(nil)
+
+func (bpr *bodyProxyReader) Read(b []byte) (int, error) {
+	return bpr.rc.Read(b)
```

Comment:
should we end the span if read returns EOF?

Reply:
or if it returns any other terminal error, should we set the span status?

Reply:
>should we end the span if read returns EOF?

I also wondered about that. It might even make more sense to end the span if it returns EOF than if Close() was invoked, because for precision, we are tracing from request sent to last byte of response received, and not necessarily on when the caller invokes response.Body.Close(). Thanks for raising this.